mne/source_tfr.py

```diff
@@ -0,0 +1,305 @@
+        be an array if there is only one source space (e.g., for volumes).
+    data : array of shape (n_dipoles, n_times)
+        The data in source space.
+    shape : tuple
+        The shape of the data. A tuple of int (n_dipoles, n_times).
+    """
+
+    @verbose
+    def __init__(self, data, vertices=None, tmin=None, tstep=None,
+                 subject=None, verbose=None):  # noqa: D102
+
+        if not (isinstance(vertices, np.ndarray) or
+                isinstance(vertices, list)):
+            raise ValueError('Vertices must be a numpy array or a list of '
+                             'arrays')
+
+        self._data_ndim = 3
+        self._src_type = 'SourceTFR'
+
+        kernel, sens_data = None, None
+        if isinstance(data, tuple):
+            if len(data) != 2:
+                raise ValueError('If data is a tuple it has to be length 2')
+            kernel, sens_data = data
+            data = None
+            if kernel.shape[1] != sens_data.shape[0]:
+                raise ValueError('kernel and sens_data have invalid '
+                                 'dimensions')
+            if sens_data.ndim != self._data_ndim:
+                raise ValueError('The sensor data must have %s dimensions, got'
+                                 ' %s' % (self._data_ndim, sens_data.ndim,))
+
+        if isinstance(vertices, list):
+            vertices = [np.asarray(v, int) for v in vertices]
+            if any(np.any(np.diff(v.astype(int)) <= 0) for v in vertices):
+                raise ValueError('Vertices must be ordered in increasing '
+                                 'order.')
+
+            n_src = sum([len(v) for v in vertices])
+
+            if len(vertices) == 1:
+                vertices = vertices[0]
+        elif isinstance(vertices, np.ndarray):
+            n_src = len(vertices)
+
+        # safeguard the user against doing something silly
+        if data is not None:
+            if data.shape[0] != n_src:
+                raise ValueError('Number of vertices (%i) and stfr.shape[0] '
+                                 '(%i) must match' % (n_src, data.shape[0]))
+            if data.ndim == self._data_ndim - 1:  # allow upbroadcasting
+                data = data[..., np.newaxis]
+            if data.ndim != self._data_ndim:
+                raise ValueError('Data (shape %s) must have %s dimensions for '
+                                 '%s' % (data.shape, self._data_ndim,
+                                         self.__class__.__name__))
+
+        self._data = data
+        self._tmin = tmin
+        self._tstep = tstep
+        self.vertices = vertices
+        self.verbose = verbose
+        self._kernel = kernel
+        self._sens_data = sens_data
+        self._kernel_removed = False
+        self._times = None
+        self._update_times()
+        self.subject = _check_subject(None, subject, False)
+
+    def __repr__(self):  # noqa: D105
```

Comment:
Don't use '%' to parse strings (we are no longer in the 90s ;) ).

fstrings are much more readable and they are the way to go ([see doc](https://www.python.org/dev/peps/pep-0498/)). However, we cannot use them until we drop python 3.5 support. Meanwhile, we could be using 'my name is {name}'.format(name='massich')

https://realpython.com/python-f-strings/

Reply:
We still use `%` all over the code base and have not agreed to change to / prefer `.format` -- I think `%` is fine for now